nilearn/input_data/masker_validation.py

```diff
@@ -0,0 +1,43 @@
+import warnings
+
+from .._utils.class_inspect import get_params
+from . import MultiNiftiMasker, NiftiMasker
+
+
+def check_embedded_nifti_masker(instance, multi=True):
+    """Base class for decomposition estimator. Takes care of mask logic and of transformation
```

Comment:
You need to be a bit more explicit about what this does:

- This creates a masker object if it does not exist
- Propagates the parameters (list them explicitly) to this masker.